Add Firestore list subcollections example

sdlin opened the pull request:
From the sdk documentation:
https://googleapis.dev/python/firestore/latest/document.html#google.cloud.firestore_v1.document.DocumentReference.collections

I ran this snippet locally:
```
>>> def list_document_subcollections(db):
...     # [START fs_list_document_subcollections]
...     collections = db.collection('cities').document('SF').collections()
...     for collection in collections:
...         for doc in collection.stream():
...             print(f'{doc.id} => {doc.to_dict()}')
...     # [END fs_list_document_subcollections]
... 
>>> list_document_subcollections(db)

example => {'foo': 'bar'}
```

busunkim96 commented on firestore/cloud-client/snippets.py:
```diff
@@ -947,3 +947,11 @@ def update_document_increment(db):
     washington_ref.update({"population": firestore.Increment(50)})
     # [END fs_update_document_increment]
+
+def list_document_subcollections(db):
```
Could you add a test for this snippet? https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/firestore/cloud-client/snippets_test.py

sdlin replied:
Yup, good call.  Added.

sdlin commented:
I had a lint issue (missing new-line) :(

Can someone please re-trigger the CI?

BenWhitehead commented:
Looks like there may be a failure in the tests @sdlin 

```
=================================== FAILURES ===================================
______________________ test_list_document_subcollections _______________________
Traceback (most recent call last):
  File "/workspace/firestore/cloud-client/snippets_test.py", line 322, in test_list_document_subcollections
    snippets.list_document_subcollections(db)
  File "/workspace/firestore/cloud-client/snippets.py", line 959, in list_document_subcollections
    collections = db.collection('cities').document('SF').collections()
AttributeError: 'function' object has no attribute 'collection'
```

sdlin commented:
> Looks like there may be a failure in the tests @sdlin
> 
> ```
> =================================== FAILURES ===================================
> ______________________ test_list_document_subcollections _______________________
> Traceback (most recent call last):
>   File "/workspace/firestore/cloud-client/snippets_test.py", line 322, in test_list_document_subcollections
>     snippets.list_document_subcollections(db)
>   File "/workspace/firestore/cloud-client/snippets.py", line 959, in list_document_subcollections
>     collections = db.collection('cities').document('SF').collections()
> AttributeError: 'function' object has no attribute 'collection'
> ```

My bad, I wasn't quite sure how the pytest db fixture got invoked.  In any case, most of the snippets just initialize the db directly and don't pass it in from the test.  I updated the snippet I added to do the same.

sdlin commented:
Sorry, I thought I fixed that newline Lint issue.

sdlin commented:
I just merged master into this branch because I thought that was the reason I wasn't allowed to merge.  Actually I think I just don't have permission to merge.

In any case, the CI needs to be run again because of the update.  Sorry about that.
